bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/discovery/BroadlinkThermostatDiscoveryService.java

```diff
@@ -0,0 +1,173 @@
+
+    private final NetworkAddressService networkAddressService;
+
+    private static final Set<ThingTypeUID> DISCOVERABLE_THING_TYPES_UIDS = Set.of(FLOUREON_THERMOSTAT_THING_TYPE,
+            UNKNOWN_BROADLINKTHERMOSTAT_THING_TYPE);
+    private static final int DISCOVERY_TIMEOUT = 30;
+
+    @Activate
+    public BroadlinkThermostatDiscoveryService(@Reference NetworkAddressService networkAddressService) {
+        super(DISCOVERABLE_THING_TYPES_UIDS, DISCOVERY_TIMEOUT);
+        this.networkAddressService = networkAddressService;
+    }
+
+    @Override
+    protected void startScan() {
+        BLDevice[] blDevices = new BLDevice[0];
+        try {
+            InetAddress sourceAddress = getIpAddress();
+            if (sourceAddress != null) {
+                logger.debug("Using source address {} for sending out broadcast request.", sourceAddress);
+                blDevices = BLDevice.discoverDevices(sourceAddress, 0, DISCOVERY_TIMEOUT * 1000);
+            } else {
+                blDevices = BLDevice.discoverDevices(DISCOVERY_TIMEOUT * 1000);
+            }
+
+        } catch (IOException e) {
+            logger.debug("Error while trying to discover broadlinkthermostat devices: ", e);
+        }
+        logger.debug("Discovery service found {} broadlinkthermostat devices.", blDevices.length);
+
+        for (BLDevice dev : blDevices) {
+            logger.debug("Broadlinkthermostat device {} of type {} with Host {} and MAC {}", dev.getDeviceDescription(),
+                    Integer.toHexString(dev.getDeviceType()), dev.getHost(), dev.getMac());
+
+            ThingUID thingUID;
+            String id = dev.getHost().replaceAll("\\.", "-");
+            logger.debug("Device ID with IP address replacement: {}", id);
+            try {
+                id = getHostnameWithoutDomain(InetAddress.getByName(dev.getHost()).getHostName());
+                logger.debug("Device ID with DNS name: {}", id);
+            } catch (UnknownHostException e) {
+                logger.debug("Discovered device with IP {} does not have a DNS name, using IP as thing UID.",
+                        dev.getHost());
+            }
+
+            switch (dev.getDeviceDescription()) {
+                case "Floureon Thermostat":
+                    thingUID = new ThingUID(FLOUREON_THERMOSTAT_THING_TYPE, id);
+                    break;
+                case "Hysen Thermostat":
+                    thingUID = new ThingUID(HYSEN_THERMOSTAT_THING_TYPE, id);
+                    break;
+                default:
+                    thingUID = new ThingUID(UNKNOWN_BROADLINKTHERMOSTAT_THING_TYPE, id);
+            }
+
+            Map<String, Object> properties = new HashMap<>();
+            properties.put(BroadlinkThermostatBindingConstants.HOST, dev.getHost());
+            properties.put(BroadlinkThermostatBindingConstants.MAC, dev.getMac().getMacString());
```

Comment:
There's a predefined constant for the mac address:
```suggestion
            properties.put(Thing.PROPERTY_MAC_ADDRESS, dev.getMac().getMacString());
```

Reply:
applied.